File: 2024/Java/Day2Part2.java
import java.io.*;
import java.nio.file.*;
import java.util.Arrays;

public class Day2Part2 {
    public static void main(String[] args) throws IOException {
        int count = 0;
        try (BufferedReader reader = Files.newBufferedReader(Path.of("2024/Input/day2.1.txt"))) {
            String line;
            while ((line = reader.readLine()) != null) {
                int[] report = Arrays.stream(line.split(" ")).mapToInt(Integer::parseInt).toArray();
                if (Utils.isSafe(report)) {
                    count++;
                } else {
                    for (int i = 0; i <  report.length; i++) {
                        int[] modifiedReport = removeElement(report, i);
                        if (Utils.isSafe(modifiedReport)) {
                            count++;
                            break;
                        }
                    }
                }
            }
        }
        System.out.println(count);
    }

    private static int[] removeElement(int[] report, int index) {
        if (report.length <= 0)
            return new int[0];

        int[] newReport = new int[report.length - 1];
        for (int i = 0, j = 0; i < report.length; i++) {
            if (i != index) {
                newReport[j++] = report[i];
            }
        }
        return newReport;
    }
}

File: 2024/Python/day2part2.py
import utils

count = 0;
report = []
file = open("2024/Input/day2.1.txt", "r")
for line in file:
    line = line.strip()
    report = list(map(int, line.split(" ")))
    if utils.isSafe(report):
        count = count + 1
    else:
        for i in range(len(report)):
            modifiedReport = report[:i] + report[i+1:]
            if utils.isSafe(modifiedReport):
                count += 1
                break
print(count)        